org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/PersistentModel.java

```diff
@@ -8,34 +8,77 @@
 
 package org.eclipse.buildship.core.preferences;
 
+import java.util.Collection;
+
+import com.google.common.base.Optional;
+
+import org.eclipse.core.runtime.IPath;
+
 /**
  * Interface to load and store Gradle model elements stored in the workspace plugin state area. The
- * model elements are stored and retrieved in a map-like fashion.
+ * model elements can be stored and retrieved in either a structured or in a map-like fashion.
  *
  * @author Donat Csikos
  */
 public interface PersistentModel {
 
     /**
+     * Returns the build directory.
+     * @return the build directory
+     */
+    Optional<IPath> getBuildDir();
```

Comment:
I've managed to change the interface such that all methods are now strongly typed.